test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,182 @@
+		framework.ExpectEqual(foundCreatedEvent, true, "unable to find the test event")
+
+		ginkgo.By("listing all events in framework namespace")
+		// get a list of Events in framework namespace to ensure endpoint coverage
+		foundCreatedEvent = listEvents(eventTestName, f.Namespace.Name, client)
+		framework.ExpectEqual(foundCreatedEvent, true, "unable to find the test event")
+
+		ginkgo.By("patching the test event")
+		// patch the event's note
+		eventPatchNote := "This is a test event - patched"
+		eventPatch, err := json.Marshal(map[string]interface{}{
+			"note": eventPatchNote,
+		})
+		framework.ExpectNoError(err, "failed to marshal the patch JSON payload")
+
+		_, err = client.Patch(context.TODO(), eventTestName, types.StrategicMergePatchType, []byte(eventPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch the test event")
+
+		ginkgo.By("getting the test event")
+		// get event by name
+		event, err := client.Get(context.TODO(), eventTestName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch the test event")
+		framework.ExpectEqual(event.Note, eventPatchNote, "test event note does not match patch note")
+		framework.ExpectEqual(event.Reason, "Test", "test event reason should not be changed")
+		framework.ExpectEqual(event.Type, "Normal", "test event type should not be changed")
+		framework.ExpectEqual(event.ReportingController, "test-controller", "test event reportingController should not be changed")
+		framework.ExpectEqual(event.ReportingInstance, "test-node", "test event reportingInstance should not be changed")
+		framework.ExpectEqual(event.ObjectMeta.Name, eventTestName, "test event name should not be changed")
+
+		ginkgo.By("deleting the test event")
+		// delete original event
+		err = client.Delete(context.TODO(), eventTestName, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "failed to delete the test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespace
+		foundCreatedEvent = listEvents(eventTestName, f.Namespace.Name, clientAllNamespaces)
+		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+
+		ginkgo.By("listing all events in framework namespace")
+		// get a list of Events in framework namespace
+		foundCreatedEvent = listEvents(eventTestName, f.Namespace.Name, client)
+		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+	})
+
+	ginkgo.It("should delete a collection of events", func() {
+		eventTestNames := []string{"test-event-1", "test-event-2", "test-event-3"}
+
+		ginkgo.By("Create set of events")
+		// create a test event in test namespace
+		for _, eventTestName := range eventTestNames {
+			_, err := client.Create(context.TODO(), testEvent(eventTestName, "testevent-set"), metav1.CreateOptions{})
+			framework.ExpectNoError(err, "failed to create event")
+			framework.Logf("created %v", eventTestName)
```

Comment:
Let's remove - if it wouldn't be created the above would return an error.

Reply:
Done.